sdk/devcenter/developer-devcenter-rest/samples-dev/sampleCreateDevBox.ts

```diff
@@ -15,119 +14,89 @@ dotenv.config();
-
-  // Get all pools for the first project
-  const poolList = await client.path("/projects/{projectName}/pools", projectName).get();
-  const pools: PoolOutput[] = [];
-
-  if (isUnexpected(poolList)) {
-    throw poolList.body.error;
-  }
-
-  console.log("Iterating through pool results:");
-
-  for await (const pool of paginate(client, poolList)) {
-    const { name } = pool;
-    console.log(`Received pool "${name}"`);
-    pools.push(pool);
-  }
-
-  if (pools.length < 1) {
-    throw "No pools found.";
-  }
-
-  const firstPool = pools[0];
-
-  if (!firstPool.name) {
-    throw "Pool is missing name";
-  }
-
-  // Create a dev box with the first pool
-  const devBoxCreateParameters: DevBoxesCreateDevBoxParameters = {
-    contentType: "application/json",
-    body: { poolName: firstPool.name },
-  };
-
-  const devBoxCreateResponse = await client
-    .path(
-      "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}",
-      projectName,
-      "me",
-      "TestDevBox"
-    )
-    .put(devBoxCreateParameters);
-
-  if (isUnexpected(devBoxCreateResponse)) {
-    throw new Error(devBoxCreateResponse.body.error.message);
-  }
-
-  const devBoxCreatePoller = getLongRunningPoller(client, devBoxCreateResponse);
-  const devBoxCreateResult = await devBoxCreatePoller.pollUntilDone();
-
-  console.log(`Provisioned dev box with state ${devBoxCreateResult.body.provisioningState}.`);
-
-  // Get the connection URL to access the machine
-  const remoteConnectionResult = await client
-    .path(
-      "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/remoteConnection",
-      projectName,
-      "me",
-      "TestDevBox"
-    )
-    .get();
-  if (isUnexpected(remoteConnectionResult)) {
-    throw new Error(remoteConnectionResult.body.error.message);
-  }
-
-  console.log(`Connect using remote connection URL ${remoteConnectionResult.body.webUrl}.`);
-
-  // Tear down the machine when finished
-  const devBoxDeleteResponse = await client
-    .path(
-      "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}",
-      projectName,
-      "me",
-      "TestDevBox"
-    )
-    .delete();
-  if (isUnexpected(devBoxDeleteResponse)) {
-    throw new Error(devBoxDeleteResponse.body.error.message);
-  }
-
-  const devBoxDeletePoller = getLongRunningPoller(client, devBoxDeleteResponse);
-  await devBoxDeletePoller.pollUntilDone();
-
-  console.log(`Cleaned up dev box successfully.`);
+    // Build client and fetch required parameters
+    const endpoint = process.env.DEVCENTER_ENDPOINT || "<devcenter name>";
+    const client = createClient(endpoint, new DefaultAzureCredential());
+
+    const projectList = await client.path("/projects").get();
+    if (isUnexpected(projectList)) {
+        throw projectList.body.error;
+    }
+
+    let project: ProjectOutput = projectList.body.value[0];
+    if (project === undefined || project.name === undefined) {
+        throw "No projects found.";
```

Comment:
I strongly prefer throwing an Error (e.g. `throw new Error("oh no"`) since that lends itself to better handling on the consumer side and allows the Error to include stack trace information.

Reply:
Updated